src/util/diproche/diprocheInterface.ts

```diff
@@ -0,0 +1,87 @@
+import IssueCode from "../../checking/issueCodes";
+import { getAllIssues } from "../../vocabularyChecker/detectWrongSyntax";
+import modes from "./predicateList.json";
+/**
+ * Since the user input most likely lacks the correct predicate,
+ * it needs to be added. For easier maintainability in case
+ * of an expanding predicate list, the corresponding predicate is
+ * added with respect to the current @param mode. This can, for
+ * example correspond to the concept of "Spielwiese". The user should
+ * be able to just stay inside a mode and not have to add this mode
+ * himself. Hence, the mode needs to be read out from the browser.
+ */
+export function addPredicate(userInput: string, mode: string) {
+	for (const predicate of modes.predicates) {
+		if (predicate.valueOf() === mode.valueOf()) {
```

Comment:
What do you need these `valueOf` for?

Reply:
This was a leftover from some earlier debugging.